### PR TITLE
Improve devcontainer, add workflow doc

### PR DESCRIPTION
**Pull request by Karakatiza666**
Converted devcontainer to docker-compose to startup Redpanda automatically.

Added instructions on building and running packages in development, and preparing demos in dev environment

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,6 +1,14 @@
 {
 	"name": "DBSP",
-	"dockerFile": "Dockerfile",
+	"dockerComposeFile": [
```
Comment (lalithsuresh) [inline .devcontainer/devcontainer.json:3]: Is there a way to provide additional arguments to docker compose? If so, we can reuse the existing docker compose files, and simply say "redpanda" as an argument to only start that service.

**Comment by Karakatiza666** [comment]
I should even be able to extend existing docker-compose. Will update the PR
soon

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (Karakatiza666) [reply .devcontainer/devcontainer.json:3]: @lalithsuresh devcontainer now employs "Extend docker-compose" feature

**Review by gz — APPROVED** [review]

File: docs/docs/guides/dev-flow.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Development workflow
+
+## Develop from Devcontainer
+
+A simple way to start interacting with DBSP source code is via supplied Devcontainer. It will prepare an environment which will let you build and run all components of DBSP out-of-the-box.
```
Comment (lalithsuresh) [inline docs/docs/guides/dev-flow.md:5]: * "with the Feldera source code."
* "the supplied Devcontainer"

File: docs/docs/guides/dev-flow.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+Spin up the Devcontainer, e.g. by opening **dbsp** repository in VS Code with [Dev Containers](vscode:extension/ms-vscode-remote.remote-containers) extension enabled.
```
Comment (lalithsuresh) [inline docs/docs/guides/dev-flow.md:7]: opening the Feldera repository...

File: docs/docs/guides/dev-flow.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+sudo chown -R user /workspaces/dbsp
+```
+
+Build the SQL Compiler:
+
+```bash
+mvn -f ./sql-to-dbsp-compiler/SQL-compiler -DskipTests package
+```
+
+Build and startup the Pipeline Manager (that also serves the DBSP Web Console):
```
Comment (lalithsuresh) [inline docs/docs/guides/dev-flow.md:20]: DBSP -> Feldera

File: docs/docs/guides/dev-flow.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+Refer to the Get Started page for basic instructions on spinning up and interacting with the demos from a separate docker-compose.
+
+## Manually starting the demos
+
+You can prepare and run multiple demos. To prepare a demo, navigate to its directory, e.g.
+```bash
+cd demo/project_demo00-SecOps
+```
+
+If `simulator` directory exists - the following initializes simulated input data and creates Kafka input and output topics:
```
Comment (lalithsuresh) [inline docs/docs/guides/dev-flow.md:57]: "If the `simulator` directory, the following command generates simulated input data and creates the required Kafka input and output topics"

File: docs/docs/guides/dev-flow.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+Now you can proceed with the [demo](#manually-starting-the-demos).
+
+## Develop on your machine
+
+TODO
+
+## Launch the prepared demo
+
+Refer to the Get Started page for basic instructions on spinning up and interacting with the demos from a separate docker-compose.
```
Comment (lalithsuresh) [inline docs/docs/guides/dev-flow.md:48]: Link to "Get Started" page

File: docs/docs/guides/dev-flow.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+```bash
+cd demo/project_demo00-SecOps
+```
+
+If `simulator` directory exists - the following initializes simulated input data and creates Kafka input and output topics:
+```bash
+cargo run --manifest-path simulator/Cargo.toml --release -- 300000
+```
+
+> For SecOps demo, argument `300000` specifies number of mock rows to be generated, which impacts duration of the demo and system load. YMMV!
```
Comment (lalithsuresh) [inline docs/docs/guides/dev-flow.md:62]: For the SecOps demo, the argument `30000` specifies the number of simulated CI pipelines to be generated....

**Review by lalithsuresh — APPROVED** [review]

**Review by blp — APPROVED** [review]
I looked at the Dockerfile a bit. I'm not an expert on Docker, but it seems reasonable to me.

I see that @lalithsuresh went through the rest of it and gave some detailed feedback, and so I didn't look at that myself.

**Pull request merged**